lock_api/src/rwlock.rs

```diff
@@ -796,7 +808,9 @@ impl<'a, R: RawRwLock + 'a, T: ?Sized + 'a> RwLockReadGuard<'a, R, T> {
     ///
     /// This is safe because `&mut` guarantees that there exist no other
     /// references to the data protected by the `RwLock`.
-    #[cfg(not(feature = "i-am-libstd"))]
+    #[cfg(not(feature = "i-am-libstd"))] // REVIEW: why the `i-am-libstd` check here?
+    // REVIEW: I'm noticing that this is probably in a number of locations, but
+    // it's not clear why various methods are configured out in that build.
```

Comment:
Because I can't figure out how to make the `defer!` macro available to this namespace without exporting it to all of libstd, which I want to avoid. Someone on IRC said it's not possible to refer to macros in your own crate by the full path.

Reply:
AFAIK the `defer!` macro is just sort of like an anonymous destructor, so could that just be made into an API rather than a macro?

You can also scope a `macro_rules` macro to just a few modules if it's defined in the crate by just putting its definition before the modules necessary